dask_expr/_merge.py

```diff
@@ -108,6 +110,45 @@ def _filter_passthrough_available(self, parent, dependents):
             return True
         return False
 
+    def _predicate_columns(self, predicate):
+        if isinstance(predicate, (Projection, Unaryop, Isin)):
+            return self._walk_predicates(predicate)
+        elif isinstance(predicate, Binop):
+            if isinstance(predicate, And):
+                return None
+
+            if not isinstance(predicate.right, Expr):
+                return self._walk_predicates(predicate.left)
+            elif isinstance(predicate.right, Elemwise):
+                return self._walk_predicates(predicate)
+            else:
+                return None
+        else:
+            # Unsupported predicate type
+            return None
+
+    def _walk_predicates(self, predicate):
```

Comment:
nit: the method names doesn't really tell what's going on. To understand what's going on one has to read and understand the entire method which I think is not necessary with a better name.

Even `_predicate_columns_inner` or smth like this would be better I think